File: 2324-find-triangular-sum-of-an-array/find-triangular-sum-of-an-array.cpp
class Solution {
public:
    int triangularSum(vector<int>& nums) {
        while (nums.size() > 1) {
            vector<int> result;
            for (int i = 0; i < nums.size() - 1; i++) {
                result.push_back((nums[i] + nums[i + 1]) % 10);
            }
            nums = result; 
        }
        return nums[0];
    }
};

File: 2324-find-triangular-sum-of-an-array/find-triangular-sum-of-an-array.java
class Solution {
    public int triangularSum(int[] nums) {
        while (nums.length > 1) {
            int[] next = new int[nums.length - 1];
            
            for (int i = 0; i < nums.length - 1; i++) {
                next[i] = (nums[i] + nums[i + 1]) % 10;
            }
            
            nums = next; 
        }
        
        return nums[0];
    }
}